stable_baselines3/common/buffers.py

```diff
@@ -199,15 +199,15 @@ def __init__(
             )
         self.optimize_memory_usage = optimize_memory_usage
 
-        self.observations = np.zeros((self.buffer_size, self.n_envs, *self.obs_shape), dtype=observation_space.dtype)
+        self.observations = np.zeros((self.buffer_size, self.n_envs, *self.obs_shape), dtype=np.float32)
```

Comment:
only cast for float type, otherwise you will waste a lot of space when using images

Reply:
Ou yes, thanks for spotting. It seems that observations are always cast to float32 by `preprocessing.preprocess_obs()` (which is called by `BaseModel.extract_features()`) anyways. 

So I removed the cast of the observations again and only cast the action to float32 if it was float64 before.